dataservice_publisher/utils/content_negotiation.py

```diff
@@ -0,0 +1,163 @@
+import logging
+from typing import Any, List, Optional
+
+
+class InvalidMediaRangeError(ValueError):
+    """Exception for invalid media ranges."""
+
+    pass
+
+
+class WeightedMediaRange:
+    """Class for handling weighted media ranges."""
+
+    type: str
+    sub_type: str
+    q: float = 1.0
+
+    def __init__(self, media_range: str) -> None:
+        """Initialize the weighted media range."""
+        weighted_media_range_split = media_range.split(";")
+        # Instantiate weighted media range:
+        try:
+            logging.debug(
+                f"Assigning q-parameter for weighted media range: {media_range}"
+            )
+            self.type, self.sub_type = weighted_media_range_split[0].split("/")
+            # If q-parameter is present, assign it:
+            for weighted_media_range_part in weighted_media_range_split[1:]:
+                if weighted_media_range_part.startswith("q="):
+                    self.q = float(
+                        # RFC specifies only 3 decimals may be used in q value.
+                        # Must strip additional decimals so that q bonus from specificity
+                        # results in correct sorting.
+                        weighted_media_range_part.split("=")[1][0:5]
+                    )
+        except ValueError as e:
+            raise InvalidMediaRangeError(f"Invalid media range: {media_range}") from e
+
+    def __eq__(self, other: Any) -> bool:  # pragma: no cover
+        """Compare two weighted media ranges."""
+        if isinstance(other, str):
+            return f"{self.type}/{self.sub_type}" == other
+        if isinstance(other, WeightedMediaRange):
+            return self.type == other.type and self.sub_type == other.sub_type
+        return False
+
+    def __str__(self) -> str:
+        """Return the weighted media range as a string."""
+        return f"{self.type}/{self.sub_type};q={self.q}"
+
+    def media_range(self) -> str:
+        """Return the media range."""
+        return f"{self.type}/{self.sub_type}"
+
+
+def prepare_weighted_media_ranges(
+    accept_weighted_media_ranges: List[str],
+) -> List[WeightedMediaRange]:
+    """Prepare the accept weighted media ranges and sort on q-parameter."""
+    logging.debug(
+        f"Preparing accept weighted media ranges: {accept_weighted_media_ranges}"
+    )
+    # Assign q-parameter:
+    accept_weighted_media_ranges_sorted: List[WeightedMediaRange] = []
+
+    for accept_weighted_media_range in accept_weighted_media_ranges:
+        # Instantiate weighted media range:
+        try:
+            weighted_media_range = WeightedMediaRange(accept_weighted_media_range)
+
+            accept_weighted_media_ranges_sorted.append(weighted_media_range)
+        except InvalidMediaRangeError:
+            logging.debug(
+                "Ignoring invalid weighted media range: %s", accept_weighted_media_range
+            )
+            pass  # ignore invalid media range
```

Comment:
Er det vanlig å bruke `pass` for å indikere at man ikke gjør noe i en `except`? Den er teoretisk overflødig, right?

Reply:
Reint estestisk, slik eg ser det. Men antyder jo intensjon, som eg synes er bra.